### PR TITLE
MRG: Fix sphere model fitting

### PR DESCRIPTION
**Pull request by larsoner**
We had a bug in our sphere model fitting. It's now fixed. While I was at it I vectorized the code so that it is much faster now.

### CODE REVIEW

File: mne/bem.py [hunk 1]
```diff
@@ -615,39 +615,33 @@ def _fwd_eeg_get_multi_sphere_model_coeffs(m, n_terms):
...
+    y, uu, sing, vv = _compose_linear_fitting_data(mu, u)
 
     # Compute the residuals
-    u['resi'] = u['y'].copy()
-
-    vec = np.empty(u['nfit'] - 1)
-    for p in range(u['nfit'] - 1):
-        vec[p] = np.dot(uu[:, p], u['y'])
-        for k in range(u['nterms'] - 1):
-            u['resi'][k] -= uu[k, p] * vec[p]
```
Comment (larsoner) [inline mne/bem.py]: here we have `uu[k, p]` ...

File: mne/bem.py [hunk 2]
```diff
@@ -657,24 +651,16 @@ def _one_step(mu, u):
         return 1.0
 
     # Compose the data for the linear fitting, compute SVD, then residuals
-    _compose_linear_fitting_data(mu, u)
-    u['uu'], u['sing'], u['vv'] = linalg.svd(u['M'])
-    u['resi'][:] = u['y'][:]
-    for p in range(u['nfit'] - 1):
-        dot = np.dot(u['uu'][p], u['y'])
-        for k in range(u['nterms'] - 1):
-            u['resi'][k] = u['resi'][k] - u['uu'][p, k] * dot
```
Comment (larsoner) [inline mne/bem.py]: ... and here `uu[p, k]`, i.e. backward, even though they are meant to do the same calculation. This was the bug that got us bad fit values before. Now we get ones like MNE-C gets (RV < 0.01)

File: mne/tests/test_bem.py [hunk 3]
```diff
@@ -106,7 +106,15 @@ def test_make_sphere_model():
...
     # tested more extensively e.g. in the forward and dipole code
-    bem = make_sphere_model('auto', 'auto', info)
+    with catch_logging() as log:
+        bem = make_sphere_model('auto', 'auto', info, verbose=True)
+    log = log.getvalue()
+    assert ' RV = ' in log
+    for line in log.split('\n'):
+        if ' RV = ' in line:
+            val = float(line.split()[-2])
+            assert val < 0.01  # actually decent fitting
```
Comment (larsoner) [inline mne/tests/test_bem.py]: on `master` we get:
```
E               assert 0.925993 < 0.01
```

**Comment by larsoner** [comment]
FWIW this seems to be a surprisingly minor bug -- these changes don't really seem to affect the output of `make_forward_solution` in the test cases I've run.

**Comment by agramfort** [comment]
thx @larsoner

**Pull request merged**